src/mode/mod.rs

```diff
@@ -31,11 +31,14 @@ where
 {
     /// Clear the display.
     pub fn clear(&mut self) -> Result<(), DisplayError> {
-        self.set_draw_area((0, 0), self.dimensions())?;
+        let dim = self.dimensions();
+        self.set_draw_area((0, 0), dim)?;
 
-        // TODO: If const generics allows this, replace `1024` with computed W x H value for current
-        // `SIZE`.
-        self.draw(&[0u8; 1024])
+        for _ in 0..dim.0 * dim.1 {
+            self.draw(&[0; 8])?;
+        }
```

Comment:
I've got a couple of questions here. I hope you don't mind a short list:

1. Because you're sending 8 bytes in a row, shouldn't this be `0..(dim.0 * dim.1) / 8`?
2. Could we increase the slice size to 64 or 128 bytes? This will still save a decent amount of memory but will reduce the overhead of doing the CS pin setup, etc.
3. To add to (1.), I think you might need a further `/ 8` because the display's dimensions are in pixels, but a page is 8 bytes tall.

Reply:
Addressed the comments you provided. Also, I realised that the addressing mode must be set and restored, so did that too.